src/js/store/projects/selectors.ts

```diff
@@ -0,0 +1,25 @@
+import { createSelector } from 'reselect';
+
+import { AppState } from '@/store';
+import { Product } from '@/store/products/reducer';
+import { selectProduct } from '@/store/products/selectors';
+import {
+  ProjectsByProductState,
+  ProjectsState,
+} from '@/store/projects/reducer';
+
+export const selectProjectState = (appState: AppState): ProjectsState =>
+  appState.projects;
+
+export const selectProjectsByProduct = createSelector(
+  [selectProjectState, selectProduct],
+  (
+    projects: ProjectsState,
+    product: Product | null | undefined,
+  ): ProjectsByProductState | undefined => {
+    if (product) {
+      return projects[product.id];
+    }
+    return undefined;
```

Comment:
Do we want to use `undefined` here or `null`? To my mind they mean slightly different things; we've tended to use `null` for "there is no value" and `undefined` for "there is no value, yet".